packages/commonwealth/client/scripts/views/components/Comments/CommentEditor/CommentEditor.tsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import { CWText } from '../../component_kit/cw_text';
+import { User } from 'views/components/user/user';
+import Topic from 'client/scripts/models/Topic';
+import Account from 'client/scripts/models/Account';
+import app from 'state';
+import clsx from 'clsx';
+import { ReactQuillEditor } from '../../react_quill_editor';
+import { CWButton } from '../../component_kit/new_designs/cw_button';
+import { CWValidationText } from '../../component_kit/cw_validation_text';
+import BN from 'bn.js';
+import { weiToTokens, getDecimals } from 'helpers';
+import { ContentType } from 'types';
+import type { DeltaStatic } from 'quill';
+import './CommentEditor.scss';
+
+type CommentEditorProps = {
+  parentType: ContentType;
+  canComment: boolean;
+  handleSubmitComment: () => void;
+  errorMsg: string;
+  contentDelta: DeltaStatic;
+  setContentDelta: React.Dispatch<React.SetStateAction<DeltaStatic>>;
+  tokenPostingThreshold: BN;
+  activeTopic: Topic;
+  userBalance: BN;
+  disabled: boolean;
+  cancel: (e: any) => void;
+  isAdmin: boolean;
+  author: Account;
+  editorValue: string;
+  shouldFocus: boolean;
+};
+
+export const CommentEditor = ({
+  parentType,
+  canComment,
+  handleSubmitComment,
+  errorMsg,
+  contentDelta,
+  setContentDelta,
+  tokenPostingThreshold,
+  activeTopic,
+  userBalance,
+  disabled,
+  cancel,
+  isAdmin,
```

Comment:
`isAdmin` is used to disable the submit button?

can we update it to be more explicit like `canSubmit`?

```
disabled={disabled && !isAdmin}
```
=> 
```
disabled={disabled && !canSubmit}
```

Reply:
I think these were split apart to indicate that whether a button is disabled is dependent both on the state of the editor (e.g. it's empty) and the user's permissions.